Library/Homebrew/test/test_utils.rb

```diff
@@ -210,4 +210,17 @@ def test_number_readable
     assert_equal "1,000", number_readable(1_000)
     assert_equal "1,000,000", number_readable(1_000_000)
   end
+
+  def test_truncate_text_to_approximate_size
+    glue = "\n[...snip...]\n" # hard-coded copy from truncate_text_to_approximate_size
+    n = 20
+    long_s = "x" * 40
+    s = truncate_text_to_approximate_size(long_s, n)
+    assert_equal n, s.length
+    assert_match(/^x+#{Regexp.escape(glue)}x+$/, s)
+    s = truncate_text_to_approximate_size(long_s, n, :front_weight => 0.0)
+    assert_equal glue + ("x" * (n - glue.length)), s
+    s = truncate_text_to_approximate_size(long_s, n, :front_weight => 1.0)
+    assert_equal ("x" * (n - glue.length)) + glue, s
```

Comment:
Yep! I'm certainly not the authority on our style rules, but my interpretation is: Use parentheses when necessary and avoid them when that's unambiguously possible and is unlikely to cause confusion for the reader.

Personally, I'm in favor of always using parentheses (probably due to my experience with other programming languages and because I don't like the inconsistency of doing the same thing in two different ways, just like `map` vs. `collect`). I do appreciate the beauty of DSLs made possible by this (like our formula DSL) and a few other use cases.